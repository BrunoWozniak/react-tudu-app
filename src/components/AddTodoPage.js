import React from 'react';
import { connect } from 'react-redux';
import TodoForm from './TodoForm';
import { startAddTodo } from '../actions/todos';


export class AddTodoPage extends React.Component {
    onSubmit = (todo) => {
        this.props.startAddTodo(todo);
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add a Todo</h1>

                    </div>
                </div>
                <div className="content-container">
                    <TodoForm
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
    startAddTodo: (todo) => dispatch(startAddTodo(todo))
});

export default connect(undefined, mapDispatchToProps)(AddTodoPage);