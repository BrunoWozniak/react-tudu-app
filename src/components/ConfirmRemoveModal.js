import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';

export default class ConfirmRemoveModal extends React.Component {
    constructor(props) {
        super(props);
    }
    
    render () {
        return(
            <Modal
                isOpen={!!this.props.confirmDelete}
                contentLabel="Are you sure?"
                closeTimeoutMS={200}
                className="modal"
            >
                <h3 className="modal__title">Are you sure?</h3>
                <p className="modal__body">Please confirm you want to delete this expense.</p>
                <div className="modal__buttons">
                    <button className="button" id="btnRemoveConfirm" onClick={this.props.onConfirmDelete}> Confirm </button>
                    <button className="button button--secondary" id="btnRemoveCancel" onClick={this.props.onCancelDelete} > Cancel </button>
                </div>
            </Modal>
        );
    }
}