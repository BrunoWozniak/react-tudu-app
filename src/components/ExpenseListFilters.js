import React from 'react';
import { connect } from 'react-redux';
import 'react-dates/initialize';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDateFilter, setEndDateFilter } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  }
  
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }
  
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  }
  
  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  }

  onShowAll = () => {
    this.props.setStartDate(null);
    this.props.setEndDate(null);
    this.props.setTextFilter('');
  }
  
  render() {
    return (
      <div className="content-container">
        <div className="input-group">
          <div className="input-group__item">
            <input
              type="text"
              className="text-input"
              placeholder="Search expenses"
              value={this.props.filters.text}
              onChange={this.onTextChange}
            />
          </div>
          <div className="input-group__item">
            <select
              className="select"
              value={this.props.filters.sortBy}
              onChange={this.onSortChange}
            >
              <option value="date">Date</option>
              <option value="amount">Amount</option>
            </select>
          </div>
          <div className="input-group__item">
            <DateRangePicker
              startDate={this.props.filters.startDate}
              endDate={this.props.filters.endDate}
              onDatesChange={this.onDatesChange}
              focusedInput={this.state.calendarFocused}
              onFocusChange={this.onFocusChange}
              showClearDates={true}
              numberOfMonths={1}
              isOutsideRange={() => false}
            />
          </div>
          <div>
            <button
              className="button button--filter-bar"
              onClick={this.onShowAll}
            >
              Show all
            </button>
          </div>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  filters: state.filters
});

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDateFilter(startDate)),
  setEndDate: (endDate) => dispatch(setEndDateFilter(endDate))
});

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
