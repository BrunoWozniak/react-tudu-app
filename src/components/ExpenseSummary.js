import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({ numberOfExpenses, totalAmount, totalNumberOfExpenses }) => {
    const expenseWord = numberOfExpenses > 1 ? 's' : '';
    const formattedTotal = numeral(totalAmount / 100).format('$0,0.00')
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <span>{numberOfExpenses}</span> expense{expenseWord} of <span>{totalNumberOfExpenses}</span> totalling <span>{formattedTotal}</span>
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add expense</Link>
                </div>
            </div>
        </div>
    )
};

const mapStateToProps = (state) => {
    return {
        numberOfExpenses: selectExpenses(state.expenses, state.filters).length,
        totalAmount: expensesTotal(selectExpenses(state.expenses, state.filters)),
        totalNumberOfExpenses: state.expenses.length
    };
};

export default connect(mapStateToProps)(ExpenseSummary);