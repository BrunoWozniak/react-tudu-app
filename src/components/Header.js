import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startSignout } from '../actions/auth';

export const Header = ({ startSignout }) => (
	<header className="header">
		<div className="content-container">
			<div className="header__content">
				<Link className="header__title" to="/dashboard">
					<h1>Tudu</h1>
				</Link>
				<button className="button button--link" onClick={startSignout}> Sign out </button>
			</div>
		</div>
	</header>
);

const mapDispatchToProps = (dispatch) => ({
	startSignout: () => dispatch(startSignout())
});

export default connect(undefined, mapDispatchToProps)(Header);