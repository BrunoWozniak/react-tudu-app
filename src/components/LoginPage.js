import React from 'react';
import { connect } from 'react-redux';
import { startSignin, startSignup } from '../actions/auth';

class LoginPage extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			email: '',
			password: '',
			formError: ''
		};
	}

	onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({ email }));
    }

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
	}
	
	onSubmitSignin = () => {
		if (!this.state.email || !this.state.password) {
            this.setState(() => ({ formError: 'Please provide email and password' }));
        } else {
			this.setState(() => ({ formError: '' }));
			this.props.startSignin({
                email: this.state.email,
                password: this.state.password
			});
		}
	}

	onSubmitSignup = (e) => {
		e.preventDefault();
		if (!this.state.email || !this.state.password) {
            this.setState(() => ({ formError: 'Please provide email and password' }));
        } else {
			this.setState(() => ({ formError: '' }));
			this.props.startSignup({
                email: this.state.email,
                password: this.state.password
			});
		}
	}

	errorMessage = () => (
		!!this.state.formError
			? this.state.formError
			: this.props.authError
				? this.props.authError
				: ''
	)
	
	render() {
		return (
			<div className="box-layout">
				<div className="box-layout__box">

					<h1 className="box-layout__title">Tudu</h1>
					<p style={{textAlign: 'center'}}>Get your todo list under control and enjoy an amazing peace of mind</p>
					<hr style={{margin: '2.0rem 0rem'}} />
			
					<div className="form">
						<div>
							<input
								name="email"
								type="email"
								className="text-input"
								placeholder="e-mail"
								style={{width: '100%'}}
								autoFocus
								onChange={this.onEmailChange}
							/>
						</div>

						<div>
							<input
								name="newPassword"
								type="password"
								className="text-input"
								placeholder="password"
								style={{width: '100%'}}	
								onChange={this.onPasswordChange}
							/>
						</div>
						<div>
							<p className="box-layout__error">{this.errorMessage()}</p>
						</div>
					</div>

					<div className="box-layout__button-box">
						<button
							className="button"
							onClick={this.onSubmitSignup}
						>
							Sign up
						</button>
						<button
							className="button"
							onClick={this.onSubmitSignin}
						>
							Sign in
						</button>
					</div>

				</div>
			</div>
		);
	}
}

const mapDispatchToProps = (dispatch) => ({
	startSignin: (signinInfo) => dispatch(startSignin(signinInfo)),
	startSignup: (signupInfo) => dispatch(startSignup(signupInfo))
});

const mapStateToProps = (state) => ({
		authError: state.auth.authError,
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);