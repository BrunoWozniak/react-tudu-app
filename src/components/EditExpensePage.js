import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import ConfirmRemoveModal from './ConfirmRemoveModal';

export class EditExpensePage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            confirmDelete: props.confirmDelete
        };
    };

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    }

    onRemove = () => {
        this.setState({ confirmDelete: true });
    }

    onConfirmDelete = (expense) => {
        this.setState({ confirmDelete: false });
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    }

    onCancelDelete = () => {
        this.setState({ confirmDelete: false })
    }
    
    render() {
        return(
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onRemove}>
                        Remove Expense
                    </button>
                </div>
                <ConfirmRemoveModal
                    confirmDelete={this.state.confirmDelete}
                    onCancelDelete={this.onCancelDelete}
                    onConfirmDelete={this.onConfirmDelete}
                />
            </div>
        )
    }
}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),
    confirmDelete: undefined
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);