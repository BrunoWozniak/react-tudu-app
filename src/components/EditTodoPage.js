import React from 'react';
import { connect } from 'react-redux';
import TodoForm from './TodoForm';
import { startEditTodo, startRemoveTodo } from '../actions/todos';
import ConfirmRemoveModal from './ConfirmRemoveModal';

export class EditTodoPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            confirmDelete: props.confirmDelete
        };
    };

    onSubmit = (todo) => {
        this.props.startEditTodo(this.props.todo.id, todo);
        this.props.history.push('/');
    }

    onRemove = () => {
        this.setState({ confirmDelete: true });
    }

    onConfirmDelete = (todo) => {
        this.setState({ confirmDelete: false });
        this.props.startRemoveTodo({ id: this.props.todo.id });
        this.props.history.push('/');
    }

    onCancelDelete = () => {
        this.setState({ confirmDelete: false })
    }
    
    render() {
        return(
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Todo</h1>
                    </div>
                </div>
                <div className="content-container">
                    <TodoForm
                        todo={this.props.todo}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onRemove}>
                        Remove Todo
                    </button>
                </div>
                <ConfirmRemoveModal
                    confirmDelete={this.state.confirmDelete}
                    onCancelDelete={this.onCancelDelete}
                    onConfirmDelete={this.onConfirmDelete}
                />
            </div>
        )
    }
}

const mapStateToProps = (state, props) => ({
    todo: state.todos.find((todo) => todo.id === props.match.params.id),
    confirmDelete: undefined
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditTodo: (id, todo) => dispatch(startEditTodo(id, todo)),
    startRemoveTodo: (id) => dispatch(startRemoveTodo(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditTodoPage);