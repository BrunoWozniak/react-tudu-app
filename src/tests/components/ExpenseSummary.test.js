import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with 2 expenses', () => {
    const numberOfExpense = 2;
    const totalAmount = 123450;
    const wrapper = shallow(
        <ExpenseSummary
            numberOfExpenses={numberOfExpense}
            totalAmount={totalAmount}
        />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with 1 expense', () => {
    const numberOfExpense = 1;
    const totalAmount = 123450;
    const wrapper = shallow(
        <ExpenseSummary
            numberOfExpenses={numberOfExpense}
            totalAmount={totalAmount}
        />);
    expect(wrapper).toMatchSnapshot();
});