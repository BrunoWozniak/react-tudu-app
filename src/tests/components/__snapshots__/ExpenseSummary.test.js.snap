// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       expense
       of 
      <span />
       totalling 
      <span>
        $1,234.50
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpenseSummary with 2 expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       expense
      s
       of 
      <span />
       totalling 
      <span>
        $1,234.50
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add expense
      </Link>
    </div>
  </div>
</div>
`;
